test/e2e/agent/agent_test.go

```diff
@@ -314,6 +314,45 @@ var _ = Describe("VM Agent behavior", func() {
 				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusUnknown))
 
 		})
+
+		It("System Info Timeout Tests", Label("81864"), func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			nextRenderedVersion, err := harness.PrepareNextDeviceVersion(deviceId)
+			Expect(err).ToNot(HaveOccurred())
+			By("Update the device image to one that has infinite.sh , timeout set to 15s")
+			_, err = harness.CheckDeviceStatus(deviceId, v1alpha1.DeviceSummaryStatusOnline)
+			Expect(err).ToNot(HaveOccurred())
+			deviceImage := fmt.Sprintf("%s/flightctl-device:v9", harness.RegistryEndpoint())
+
+			var osImageSpec = v1alpha1.DeviceOsSpec{
+				Image: deviceImage,
+			}
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+				device.Spec.Os = &osImageSpec
+
+				logrus.Infof("Updating %s with OS image", osImageSpec)
+			})
+
+			err = harness.WaitForDeviceNewRenderedVersion(deviceId, nextRenderedVersion)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Reload the flight agent")
+			_, err = harness.VM.RunSSH([]string{"sudo", "systemctl", "reload", "flightctl-agent"}, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Custom system-info infinite is empty due to timeout")
+			// wait for the device to pickup enrollment and report measurements on device status
```

Comment:
Is this ensuring that the timeout has actually elapsed before checking the custom info?

Reply:
according to the test case :
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-81864
the indication of :
```

  system-info:
    system-info-custom:
      infinite: ""
```
means that infinite has timed out , if we are using the same status to indicate lack of status / error with running then i dont see how we can differentiate between time out and anything else - not in prod and not in test

perhaps im missing something ? @kkyrazis

Reply:
That context helps a lot thanks.